PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:8 + add:0, change matches:
-    _inherit = ['barcode.rule']
-    _inherit = ['loyalty.mail']
-    _inherit = ['pos.config']
-    _inherit = ['pos.order']
-    _inherit = ['pos.order.line']
-    _inherit = ['pos.session']
-    _inherit = ['product.product']
-    _inherit = ['res.partner']

Total Changes: 20

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pos_loyalty/models/barcode_rule.py
+++ b/addons/pos_loyalty/models/barcode_rule.py
@@ -5,6 +5,6 @@ from odoo import models, fields
 
 
 class BarcodeRule(models.Model):
-    _inherit = ['barcode.rule']
+    _inherit = 'barcode.rule'
 
     type = fields.Selection(selection_add=[('coupon', 'Coupon')], ondelete={'coupon': 'set default'})

--- a/addons/pos_loyalty/models/loyalty_card.py
+++ b/addons/pos_loyalty/models/loyalty_card.py
@@ -5,6 +5,7 @@ from odoo import fields, models, api
 
 
 class LoyaltyCard(models.Model):
+    _name = 'loyalty.card'
     _inherit = ['loyalty.card', 'pos.load.mixin']
 
     source_pos_order_id = fields.Many2one('pos.order', "PoS Order Reference",

--- a/addons/pos_loyalty/models/loyalty_mail.py
+++ b/addons/pos_loyalty/models/loyalty_mail.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class LoyaltyMail(models.Model):
-    _inherit = ['loyalty.mail']
+    _inherit = 'loyalty.mail'
 
     pos_report_print_id = fields.Many2one('ir.actions.report', string="Print Report", domain=[('model', '=', 'loyalty.card')],
         help="The report action to be executed when creating a coupon/gift card/loyalty card in the PoS.",

--- a/addons/pos_loyalty/models/loyalty_program.py
+++ b/addons/pos_loyalty/models/loyalty_program.py
@@ -5,6 +5,7 @@ from odoo.exceptions import UserError
 
 
 class LoyaltyProgram(models.Model):
+    _name = 'loyalty.program'
     _inherit = ['loyalty.program', 'pos.load.mixin']
 
     # NOTE: `pos_config_ids` satisfies an excpeptional use case: when no PoS is specified, the loyalty program is

--- a/addons/pos_loyalty/models/loyalty_reward.py
+++ b/addons/pos_loyalty/models/loyalty_reward.py
@@ -7,6 +7,7 @@ import json
 
 
 class LoyaltyReward(models.Model):
+    _name = 'loyalty.reward'
     _inherit = ['loyalty.reward', 'pos.load.mixin']
 
     def _get_discount_product_values(self):

--- a/addons/pos_loyalty/models/loyalty_rule.py
+++ b/addons/pos_loyalty/models/loyalty_rule.py
@@ -6,6 +6,7 @@ from odoo.osv import expression
 
 
 class LoyaltyRule(models.Model):
+    _name = 'loyalty.rule'
     _inherit = ['loyalty.rule', 'pos.load.mixin']
 
     valid_product_ids = fields.Many2many(

--- a/addons/pos_loyalty/models/pos_config.py
+++ b/addons/pos_loyalty/models/pos_config.py
@@ -6,7 +6,7 @@ from odoo.exceptions import UserError
 
 
 class PosConfig(models.Model):
-    _inherit = ['pos.config']
+    _inherit = 'pos.config'
 
     # NOTE: this funtions acts as a m2m field with loyalty.program model. We do this to handle an excpetional use case:
     # When no PoS is specified at a loyalty program form, this program is applied to every PoS (instead of none)

--- a/addons/pos_loyalty/models/pos_order.py
+++ b/addons/pos_loyalty/models/pos_order.py
@@ -8,7 +8,7 @@ import base64
 
 
 class PosOrder(models.Model):
-    _inherit = ['pos.order']
+    _inherit = 'pos.order'
 
     def validate_coupon_programs(self, point_changes, new_codes):
         """

--- a/addons/pos_loyalty/models/pos_order_line.py
+++ b/addons/pos_loyalty/models/pos_order_line.py
@@ -5,7 +5,7 @@ from odoo import fields, models, api
 
 
 class PosOrderLine(models.Model):
-    _inherit = ['pos.order.line']
+    _inherit = 'pos.order.line'
 
     is_reward_line = fields.Boolean(
         help="Whether this line is part of a reward or not.")

--- a/addons/pos_loyalty/models/pos_session.py
+++ b/addons/pos_loyalty/models/pos_session.py
@@ -4,7 +4,7 @@ from odoo import models, api
 
 
 class PosSession(models.Model):
-    _inherit = ['pos.session']
+    _inherit = 'pos.session'
 
     @api.model
     def _load_pos_data_models(self, config_id):

--- a/addons/pos_loyalty/models/product_product.py
+++ b/addons/pos_loyalty/models/product_product.py
@@ -2,7 +2,7 @@ from odoo import api, models
 
 
 class ProductProduct(models.Model):
-    _inherit = ['product.product']
+    _inherit = 'product.product'
 
     @api.model
     def _load_pos_data_fields(self, config_id):

--- a/addons/pos_loyalty/models/res_partner.py
+++ b/addons/pos_loyalty/models/res_partner.py
@@ -2,6 +2,6 @@ from odoo import fields, models
 
 
 class ResPartner(models.Model):
-    _inherit = ['res.partner']
+    _inherit = 'res.partner'
 
     loyalty_card_count = fields.Integer(groups='base.group_user,point_of_sale.group_pos_user')
